controller/pkg/flowtracking/flowtracking.go

```diff
@@ -51,6 +69,18 @@ func (c *Client) UpdateMark(ipSrc, ipDst net.IP, protonum uint8, srcport, dstpor
 	return c.UpdateApplicationFlowMark(ipSrc, ipDst, protonum, srcport, dstport, newmark)
 }
 
+// GetOriginalDest gets the original destination ip, port and the mark on the packet
+func (c *Client) GetOriginalDest(ipSrc, ipDst net.IP, srcport, dstport uint16, protonum uint8) (net.IP, uint16, uint32, error) {
+	flow := conntrack.NewFlow(protonum, 0, ipSrc, ipDst, srcport, dstport, 0, 0)
+	origFlow, err := c.conn.Get(flow)
+
+	if err != nil {
+		zap.L().Error("")
```

Comment:
beautiful!

Reply:
So the get function doesn’t return a pointer. Even on an error it returns the 0 object. So that code is harmless. It won’t panic. I will get rid of the empty error message there because the client is handling the error. You can look at the Get function here https://github.com/ti-mo/conntrack/blob/master/conn.go